src/lib/provable/foreign-field.ts

```diff
@@ -92,6 +92,31 @@ class ForeignField {
    * ```ts
    * let x = new ForeignField(5);
    * ```
+   * 
+   * Note: Inputs must be range checked if they originate from a different field with a different modulus or if they are not constants.
+   * 
+   * - When constructing from another {@link ForeignField} instance, ensure the modulus matches. If not, check the modulus using `Gadgets.ForeignField.assertLessThan()` and handle appropriately.
+   *   @example
+   *   ```ts
+   *   class SmallField extends createForeignField(17n) {}
+   *   class LargerField extends createForeignField(19n) {}
+   * 
+   *   let smallField = new SmallField(10);
+   *   let largerField = new LargerField(17);
+   *   Gadgets.ForeignField.assertLessThan(largerField.value, SmallField.modulus);
```

Comment:
I'm not a fan of using constants (10, 17) here, because there's no need to prove that constants are reduced (and the assertion is a no-op). I think you should rather leave open how the fields are constructed, or show how they are witnessed as variables:

```ts
let smallField: SmallField = ...
```

```ts
let smallField = Provable.witness(SmallField, () => 10n);
```